_posts/2022-03-25-conformation-generation.md

```diff
@@ -0,0 +1,150 @@
+
+- [Neural ODE systems](https://arxiv.org/pdf/1806.07366.pdf): In a neural ODE, we specify an initial value problem that uses a neural network to specify the “dynamics” of the system (or the derivative of the “state” with respect to time). More concretely, we have that $y(0) = y_0$ and that $\frac{dy}{dt} = f(y(t), t, \theta)$. Using an ODE solver such as `odeint`, we can calculate the value of $y$ at any time $t$ as in any initial value problem. <br> In fact, y can be thought of as a [residual network](http://implicit-layers-tutorial.org/neural_odes/) where we take the limit with respect to the number of layers.
+
+    - Correspondingly, in this work, the purpose of instantiating an ODE is to be able to predict $d(t)$ – the distances between each pair of atoms at any time point. $\frac{d\mathbf{d}}{dt}$ can be predicted at any time point given $d(t)$, the time point, $t$, the molecular graph, and the parameters of the assigned neural network (in our case an MPNN).
+
+$$\boldsymbol{d} = F_\theta(\boldsymbol{d}(t_0), \mathcal{G}) = \boldsymbol{d}(t_0) + \int_{t_0}^{t_1} f_\theta(\boldsymbol{d}(t), t; \mathcal{G})dt$$
+
+To combine the two methods above: We take $z_0$ and define it as the initial value. $\frac{dz}{dt}$ is calculated using a neural network that takes in $z(t)$, $t$, and $\theta$. With $z_0$ and a function $f$ to calculate $\frac{dz}{dt}$ at any time point, $z(t)$ can be calculated as per the traditional initial value problem formulation. The ODESolver also predicts the $\textrm{log}(p(z(t))$ at any time point, thereby encoding the density function for $z(t)$ in addition to just the values of $z(t)$ alone (Figure 2).
+
+<p align="center">
+  <img src="{{ site.url }}/public/images/2022-03-25-conformation-generation/ode.png">
+  <b> Figure 2: </b> The neural ODE system computes $\boldsymbol{d}(t)$ and $\textrm{log}(p(\boldsymbol{d}(t))$ at various time points in order to try and approximate the actual functions for $\boldsymbol{d}(t)$ and $\textrm{log}(p(\boldsymbol{d}(t))$.
+</p>
+
+In this case, our $z(t)$ is $\boldsymbol{d}(t)$, a function that outputs a vector with pairwise intramolecular distances. The “continuous-time dynamics" is a function that takes in neural network parameters, the time, and the current state to output the derivative of the distances with respect to time. The neural network is a graph [message passing neural network](https://paperswithcode.com/method/mpnn) (MPNN) that calculates node and edge representations and aggregates the node and edge representations for each bond to calculate $\frac{dd_{uv}}{dt}$ – the change of the distance between two atoms with respect to time (Figure 3).
+
+<p align="center">
+  <img src="/public/images/2021-12-01-conformation-generation/mpnn.png">
```

Comment:
can you update this one to `{{ site.url }}/ages/2022-03-25-conformation-generation/mpnn.png"`?